docs/golang/README.md

```diff
@@ -1,3 +1,182 @@
-# Golang ssb
+# go-ssb
+
+## Introdcution
+
+Go-SSB implements most of the existing javascript functionallity but is less modular, due to the typed nautre of the language. Most of the code lives in a single repository: [go-ssb](https://https://github.com/cryptoscope/ssb) with a couple of packages split out, like `go-muxrpc` and `secretstream`.
+
+The import path for the code is `go.cryptoscope.co/ssb`.
+
+If you are not familiar with the [Go programming language](https://golang.org), I suggest you get started with the [guided tour](https://tour.golang.org/welcome/1).
+
+
+## Packages Overview
+
+This should serve as guardrails when looking into the repo for the first time.
+
+### cmd
+
+By coding convention, this contains the executable tools. Most importantly these two:
+
+* `go-sbot` which hosts the database and p2p server.
+* `sbotcli` a muxrpc client to do common tasks, like publishing a message or querying messages.
+
+### Message
+
+Message contains all the code to create and verify messages.
+
+The current established format is
+
+* message - abstract verification and publish helpers
+* message/legacy - how to encode and verify a ssb-v1 message
+* message/multimsg - encoding multiple kinds of messages to disk (see _Database Concept Overview_ below)
+
+### Plugins
+
+Plugins supply muxrpc handlers to query the database in certain ways.
+They also implement other sub-protocols like ssb-blobs.
+
+* gossip - implements the `createHistoryStream` muxrpc call. _Legacy_ (non-EBT) Replication of how to fetch a bunch of feeds is found here.
+* replicate - roughly translates to [ssb-replicate](https://github.com/ssbc/ssb-replicate), controlls which feeds to fetch
+* control - connect to remote peers. (a naming hack, supplies `ctrl.connect` which should actually be `gossip.connect`)
+* status - return some information about the running server (number of connected peers, number of messages, index state)
+* publish - exposes `sbot.PublishLog.Publish(...)` as the muxrpc async call `publish`
+* get - very short module, just exposes [sbot.Get](https://pkg.go.dev/go.cryptoscope.co/ssb/sbot#Sbot.Get) over muxrpc
+* rawread - a bunch of _read messages from the Receive Log_ functions likes (`createLogStream` and `messagesByType`)
+* legacyinvites - the first invite system [ssb-invite](https://github.com/ssbc/ssb-invite/)
+* peerinvites - the server part of the newer invite system: [ssb-peer-invites](https://github.com/ssbc/ssb-peer-invites)
+* private - about to be deprecated way of accessing private messages
+* friends - supplies some of [ssb-friends](https://github.com/ssbc/ssb-friends)
```

Comment:
supplies some of the functionality of 

Probabably better to write what parts?